### PR TITLE
[chore] Upgrade pymongo dep version to 4.6.3

### PR DESCRIPTION
**Pull request by crobert-1**

### CODE REVIEW

**Review by atoulme — APPROVED** [review]

File: internal/signalfx-agent/bundle/collectd-plugins.yaml [hunk 1]
```diff
@@ -31,7 +31,7 @@
   version: v1.3.7
   repo: signalfx/collectd-mongodb
   pip_packages:
-    - pymongo[tls]==3.13.0
+    - pymongo[tls]==4.6.3
```
Comment (jinja2) [inline internal/signalfx-agent/bundle/collectd-plugins.yaml:34]: This major version update might be breaking for the collectd-mongodb plugin. For e.g., [v4.x replaces the parameter names for ssl args](https://pymongo.readthedocs.io/en/stable/migrate-to-pymongo4.html#renamed-uri-options) which we seem to be using in [our plugin](https://github.com/signalfx/collectd-mongodb/blob/master/mongodb.py#L88-L95).

Comment (atoulme) [reply internal/signalfx-agent/bundle/collectd-plugins.yaml:34]: ah, I assumed we had an integration test covering this.

Comment (crobert-1) [reply internal/signalfx-agent/bundle/collectd-plugins.yaml:34]: Good catch! I'll look into this 👍

**Comment by crobert-1** [comment]
Superseded by https://github.com/signalfx/splunk-otel-collector/pull/4731

**Pull request closed without merging**